l1-contracts/contracts/bridge/ntv/L2NativeTokenVault.sol

```diff
@@ -125,13 +125,34 @@ contract L2NativeTokenVault is IL2NativeTokenVault, NativeTokenVault {
         }
     }
 
-    /// @notice Deploys the beacon proxy for the L2 token, while using ContractDeployer system contract.
+    /// @notice Ensures that the token is deployed inner for legacy tokens. 
+    function _ensureTokenDeployedInnerLegacyToken(
+        bytes32 _assetId,
+        address _originToken,
+        address _expectedToken,
+        address _l1LegacyToken
+    ) internal {
+        _assetIdCheck(L1_CHAIN_ID, _assetId, _originToken);
+
+        /// token is a legacy token, no need to deploy
+        if (_l1LegacyToken != _originToken) {
+            revert AddressMismatch(_originToken, _l1LegacyToken);
+        }
```

Comment:
I think from a style suggestion this is actually better, because the two ensureTokenDeployedInners are very similar: 
<img width="1593" alt="image" src="https://github.com/user-attachments/assets/0f5592db-a4c8-431e-a934-545e7f20bf91">

Reply:
This is nit, so can leave as it is, but "decoding" was confusing ¯\_(ツ)_/¯

Reply:
I mean, but then comparing the two is easier, I'll leave it as it is for now